lib/mocha.js

```diff
@@ -277,15 +283,67 @@ Mocha.prototype.invert = function() {
  * @param {Boolean} ignore
  * @return {Mocha}
  * @api public
- * @param {boolean} ignore
- * @return {Mocha}
  */
 Mocha.prototype.ignoreLeaks = function(ignore) {
   this.options.ignoreLeaks = Boolean(ignore);
   return this;
 };
 
 /**
+ * Enable or disable randomization of test execution order within suites.
+ *
+ * @param {(boolean|number|string)} [seed] Optional random seed.  Seed must be
+ *   a 32-bit unsigned integer, or a string which can convert to one.  If
+ *   `true`, a seed will be created.  If `false`, randomization will be
+ *   globally disabled (if it was enabled previously).
+ * @return {Mocha}
+ * @api public
+ */
+Mocha.prototype.randomize = function randomize(seed) {
+  if (seed !== false) {
+    if (!arguments.length || seed === true) {
+      // generate a number for seeding
+      seed = Mocha.seed();
+    } else if (!(typeof seed === 'number' && seed > 0
+      && seed <= MAX_SAFE_INTEGER && seed % 1 === 0)) {
+      // this would be a hexadecimal number represented as a string
+      if (/^(0[xX])?[a-fA-F0-9]{1,14}$'/.test(seed)) {
+        seed = parseInt(seed, 16);
+      } else {
+        // a decimal number as string, presumably
+        seed = parseInt(seed, 10);
+        if (isNaN(seed)) {
+          throw new Error('Invalid random seed. Expected unsigned 32-bit '
+            + 'integer; got: "' + seed + '"');
+        }
+      }
+    }
+
+    var engine = Random.engines.mt19937().seed(seed);
```

Comment:
I wonder if quality of a PRNG really matters for Mocha. Mersenne Twister famously has a very long period although not being that strong or fast, but period is basically what we need here because we want a maximum number of test order combinations. With that in mind, sure, newer faster and altogether _better_ PRNGs are cool, but not really useful.